rs-backend/src/api.rs

```diff
@@ -0,0 +1,167 @@
+// Types implement both serialize and deserialize despite conceptually needing
+// only one for testing.
```

Comment:
Well, it's mainly just poorly phrased: should really be "Type implement both serialize and deserialize despite only conceptually needing only one due to testing requiring serialization of requests". Not sure if it's really needed... I can edit it into a more thorough explanation though, or just remove it.